### PR TITLE
Support glean apps with additional deletion requests table in shredder

### PR DESCRIPTION
**Pull request by relud**

### CODE REVIEW

File: bigquery_etl/shredder/config.py [hunk 1]
```diff
@@ -406,19 +407,48 @@ def find_glean_targets(pool, client, project=SHARED_PROD):
...
-            re.sub("_stable$", "_derived", table.dataset_id),
-        ]
-        if dataset_id in datasets
-    }
+    sources = defaultdict(default_factory=list)
+    for table in glean_stable_tables:
+        if table.table_id.startswith(source_doctype):
+            source = DeleteSource(qualified_table_id(table), GLEAN_CLIENT_ID, project)
+            derived_dataset = re.sub("_stable$", "_derived", table.dataset_id)
+            # append to list to use every version of deletion request tables
```
Comment (relud) [inline bigquery_etl/shredder/config.py:415]: in practice, we only have one version, so this shouldn't impact behavior

File: tests/shredder/test_config.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        return [
+            bigquery.DatasetReference(project, f"{app}_{suffix}")
+            for app in ["fenix", "focus_android"]
+            for suffix in ["stable", "derived"]
+        ]
+
+    def list_tables(self, dataset_ref):
+        labels = {}
+        if dataset_ref.dataset_id.endswith("stable"):
+            table_ids = ["metrics_v1", "deletion_request_v1"]
```
Comment (sean-rose) [inline tests/shredder/test_config.py:19]: For better test coverage it'd be good to have a table that starts with `migration` and verify it's excluded (i.e. no change to expected output).

File: tests/shredder/test_config.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            table.schema = [
+                bigquery.SchemaField(
+                    "client_info",
+                    "RECORD",
+                    "NULLABLE",
+                    [bigquery.SchemaField("client_id", "STRING")],
+                )
+            ]
+        elif table_ref.dataset_id.endswith("derived"):
+            table.schema = [bigquery.SchemaField("client_id", "STRING")]
```
Comment (sean-rose) [inline tests/shredder/test_config.py:51]: For better test coverage it'd be good to also have a table that doesn't have a `client_id` column and verify it's excluded (i.e. no change to expected output).

**Review by sean-rose — APPROVED** [review]
r+wc

**Pull request merged**